src/pivot-table/hooks/use-column-width/use-column-width-left.ts

```diff
@@ -0,0 +1,141 @@
+import { ColumnWidthType } from "@qlik/nebula-table-utils/lib/constants";
+import { useMeasureText, useOnPropsChange } from "@qlik/nebula-table-utils/lib/hooks";
+import { useCallback, useState } from "react";
+import { PSEUDO_DIMENSION_KEY } from "../../../constants";
+import type { HeaderCell } from "../../../types/types";
+import { useStyleContext } from "../../contexts/StyleProvider";
+import {
+  EXPAND_ICON_SIZE,
+  LEFT_GRID_MAX_WIDTH_RATIO,
+  LOCK_ICON_SIZE,
+  MENU_ICON_SIZE,
+  TOTAL_CELL_PADDING,
+} from "./constants";
+import type { ColumnWidthLeftHook, LeftGridWidthInfo } from "./types";
+import { getColumnWidthValue, getMeasureTextArgs } from "./utils";
+
+export default function useColumnWidthLeft({ layoutService, tableRect, headersData }: ColumnWidthLeftHook) {
+  const {
+    layout: {
+      qHyperCube: { qMeasureInfo, qNoOfLeftDims },
+    },
+    isFullyExpanded,
+  } = layoutService;
+  const styleService = useStyleContext();
+  const { measureText: measureTextForHeader } = useMeasureText(getMeasureTextArgs(styleService.header));
+  const { measureText: measureTextForMeasureValue } = useMeasureText(getMeasureTextArgs(styleService.measureValues));
+  const { estimateWidth: estimateWidthForDimensionValue, measureText: measureTextForDimensionValue } = useMeasureText(
+    getMeasureTextArgs(styleService.dimensionValues),
+  );
+
+  const calculateLeftGridWidthInfo = useCallback(
```

Comment:
I tried breaking some logic out of this one, but it is all very connected, and you would need to pass a lot of args